packages/backend/src/modules/authResolvers.ts

```diff
@@ -0,0 +1,66 @@
+import { OidcAuthResult } from '@backstage/plugin-auth-backend-module-oidc-provider';
+import {
+  AuthResolverContext,
+  createSignInResolverFactory,
+  OAuthAuthenticatorResult,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+
+const KEYCLOAK_ID_ANNOTATION = 'keycloak.org/id';
+const PING_IDENTITY_ID_ANNOTATION = 'pingidentity.org/id';
+
+/**
+ * Additional sign-in resolvers for the Oidc auth provider.
+ *
+ * @public
+ */
+export namespace rhdhSignInResolvers {
+  /**
+   * A oidc resolver that looks up the user using their Keycloak user ID
+   */
+  export const oidcSubClaimMatchingKeycloakUserId = createSignInResolverFactory(
+    {
+      create() {
+        return async (
+          info: SignInInfo<OAuthAuthenticatorResult<OidcAuthResult>>,
+          ctx: AuthResolverContext,
+        ) => {
+          const sub = info.result.fullProfile.userinfo.sub;
+
+          if (!sub) {
+            throw new Error(
+              `Keycloak user profile does not contain a sub claim`,
```

Comment:
The only case I can think of is when the IdP is misconfigured and does not grant permissions to return the `openid` scope. I believe the user has to exist in the IdP because the resolver is only called when the user has already been successfully authenticated.

But I do agree with the message being not very helpful, how does this sound?
`The user profile from ${providerName} is missing a 'sub' claim, likely due to a misconfiguration in the provider. Please contact your system administrator for assistance.`

Reply:
Sure, that sounds good